tests/testautonatv2.nim

```diff
@@ -0,0 +1,120 @@
+{.used.}
+
+# Nim-Libp2p
+# Copyright (c) 2025 Status Research & Development GmbH
+# Licensed under either of
+#  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+#  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+# at your option.
+# This file may not be copied, modified, or distributed except according to
+# those terms.
+
+import std/options
+import chronos
+import
+  ../libp2p/[
+    transports/tcptransport,
+    upgrademngrs/upgrade,
+    builders,
+    protocols/connectivity/autonatv2/core,
+      # nameresolving/nameresolver,
+      # nameresolving/mockresolver,
+  ],
+  ./helpers
+
+suite "AutonatV2":
+  teardown:
+    checkTrackers()
+
+  asyncTest "encode/decode messages":
+    # DialRequest
+    let dialReq = DialRequest(
+      addrs:
+        @[
+          MultiAddress.init("/ip4/127.0.0.1/tcp/4040").get(),
+          MultiAddress.init("/ip4/127.0.0.1/tcp/4041").get(),
+        ],
+      nonce: 42,
+    )
+    let encodedReq = dialReq.encode()
+    let decodedReq = DialRequest.decode(encodedReq).get()
```

Comment:
```suggestion
    let decodedReq = DialRequest.decode(dialReq.encode()).get()
```

nitpicking this here and other places in this file, because variable is not really needed

Reply:
it is also possible to avoid `decodedReq`:

```nim
check decodedReq == DialRequest.decode(dialReq.encode()).get()
```